doc/classes/@GlobalScope.xml

```diff
@@ -2404,11 +2404,51 @@
+			The number of DirectInput game controller buttons.
+		</constant>
+		<constant name="JOY_BUTTON_LEFT_X_NEG" value="128" enum="JoyButton">
+			Game controller button associated with the negative side of left joystick x-axis, when configured as button.
+		</constant>
+		<constant name="JOY_BUTTON_LEFT_X_POS" value="129" enum="JoyButton">
+			Game controller button associated with the positive side of left joystick x-axis, when configured as button.
+		</constant>
+		<constant name="JOY_BUTTON_LEFT_Y_NEG" value="130" enum="JoyButton">
+			Game controller button associated with the negative side of left joystick y-axis, when configured as button.
+		</constant>
+		<constant name="JOY_BUTTON_LEFT_Y_POS" value="131" enum="JoyButton">
+			Game controller button associated with the positive side of left joystick y-axis, when configured as button.
+		</constant>
+		<constant name="JOY_BUTTON_RIGHT_X_NEG" value="132" enum="JoyButton">
+			Game controller button associated with the negative side of right joystick x-axis, when configured as button.
+		</constant>
+		<constant name="JOY_BUTTON_RIGHT_X_POS" value="133" enum="JoyButton">
+			Game controller button associated with the positive side of right joystick x-axis, when configured as button.
+		</constant>
+		<constant name="JOY_BUTTON_RIGHT_Y_NEG" value="134" enum="JoyButton">
+			Game controller button associated with the negative side of right joystick y-axis, when configured as button.
+		</constant>
+		<constant name="JOY_BUTTON_RIGHT_Y_POS" value="135" enum="JoyButton">
+			Game controller button associated with the positive side of right joystick y-axis, when configured as button.
+		</constant>
+		<constant name="JOY_BUTTON_TRIGGER_LEFT_NEG" value="136" enum="JoyButton">
+			Game controller button associated with the negative side of left trigger axis, when configured as button.
+		</constant>
+		<constant name="JOY_BUTTON_TRIGGER_LEFT_POS" value="137" enum="JoyButton">
+			Game controller button associated with the positive side of left trigger axis, when configured as button.
+		</constant>
+		<constant name="JOY_BUTTON_TRIGGER_RIGHT_NEG" value="138" enum="JoyButton">
+			Game controller button associated with the negative side of right trigger axis, when configured as button.
+		</constant>
+		<constant name="JOY_BUTTON_TRIGGER_RIGHT_POS" value="139" enum="JoyButton">
+			Game controller button associated with the positive side of right trigger axis, when configured as button.
+		</constant>
+		<constant name="JOY_BUTTON_MAX" value="148" enum="JoyButton">
 			The maximum number of game controller buttons supported by the engine. The actual limit may be lower on specific platforms:
 			- [b]Android:[/b] Up to 36 buttons.
 			- [b]Linux:[/b] Up to 80 buttons.
 			- [b]Windows[/b] and [b]macOS:[/b] Up to 128 buttons.
+			- Buttons 128-147 are virtual buttons, in the case that the respective axis is configured to be treated as a button.
```

Comment:
I wonder if the value shouldn't be higher, in case there is ever need for more buttons. Enums have a very big range, it could even start with 1000.

Reply:
Do you want to create this gap in order to future-proof situations, where operating systems increase the maximum number of supported controller buttons? I am fine with this change.